File: apps/api/src/app/muscle-group/muscle-group.entity.ts
import { Column, Entity, OneToMany, PrimaryGeneratedColumn } from 'typeorm';
import { MuscleEntity } from '../muscle/muscle.entity';
import { WorkoutEntity } from '../workout/workout.entity';

@Entity({ name: 'muscle_group', schema: 'public' })
export class MuscleGroupEntity {
    @PrimaryGeneratedColumn('increment')
    id: number;

    @Column()
    name: string;

    @OneToMany(() => MuscleEntity, (muscle) => muscle.muscleGroup, { onDelete: 'CASCADE' })
    muscle: MuscleEntity[];

    @OneToMany(() => WorkoutEntity, workoutEntity => workoutEntity.muscleGroup, { onDelete: 'CASCADE' })
    workout: WorkoutEntity[];
}

File: my-workout-api/src/main/java/com/avernet/myworkoutapi/musclegroup/MuscleGroupEntity.java
package com.avernet.myworkoutapi.musclegroup;

import com.avernet.myworkoutapi.muscle.MuscleEntity;
import com.avernet.myworkoutapi.workout.WorkoutEntity;
import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "muscle_group")
@Getter
@Setter
public class MuscleGroupEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Integer id;

    @Enumerated(EnumType.STRING)
    @Column(name = "name")
    MuscleGroupType name;

    @OneToMany(fetch = FetchType.LAZY, mappedBy = "muscleGroup", cascade = CascadeType.ALL, orphanRemoval = true)
    List<WorkoutEntity> workouts = new ArrayList<>();

    @OneToMany(fetch = FetchType.LAZY, mappedBy = "muscleGroup", cascade = CascadeType.ALL, orphanRemoval = true)
    List<MuscleEntity> muscleList = new ArrayList<>();
}
